boa/contracts/abi/contract.py

```diff
@@ -0,0 +1,122 @@
+from functools import cached_property
+from itertools import groupby
+from os.path import basename
+from typing import Any, Optional
+
+from _operator import attrgetter
+from eth.abc import ComputationAPI
+
+from boa.contracts.abi import _decode_addresses, _format_abi_type
+from boa.contracts.abi.function import ABIFunction, ABIOverload
+from boa.contracts.evm_contract import BaseEVMContract
+from boa.environment import Address
+from boa.util.abi import abi_decode
+from boa.util.exceptions import StackTrace, _handle_child_trace
+
+
+class ABIContract(BaseEVMContract):
+    """A contract that has been deployed to the blockchain and created via an ABI."""
+
+    def __init__(
+        self,
+        name: str,
+        functions: list[ABIFunction],
+        address: Address,
+        filename: Optional[str] = None,
+        env=None,
+    ):
+        super().__init__(env, filename=filename, address=address)
+        self._name = name
+        self._functions = functions
+
+        for name, group in groupby(self._functions, key=attrgetter("name")):
```

Comment:
i'm not sure this works -- groupby requires sorting

per https://stackoverflow.com/a/75198417/

> items are only grouped together as long as they're sequential in the iterable

Reply:
Good catch. I think all the abi genetators put the overloaded functions together, but this is still incorrect